crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -545,7 +504,7 @@ unsafe impl WorldQuery for Entity {
 }
 
 impl<'w> WorldQueryGats<'w> for Entity {
-    type Fetch = EntityFetch<'w>;
+    type Fetch = EntityFetch;
```

Comment:
```suggestion
    type Fetch = ();
```
I don't think `EntityFetch` needs to exist anymore.